File: solutions/002.add-two-numbers/add-two-numbers.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var addTwoNumbers = function(l1, l2) {
    let cur1 = l1, cur2 = l2, carry = 0;
    let res = new ListNode(1), temp = res;
    while(cur1 !== null || cur2 !== null || carry !== 0){
        const num1 = (cur1 !== null) ? cur1.val : 0;
        const num2 = (cur2 !== null) ? cur2.val : 0;
        const sum = num1 + num2 + carry;
        carry = Math.floor(sum / 10);
        temp.next = new ListNode(sum % 10);
        temp = temp.next;
        if(cur1 !== null) cur1 = cur1.next;
        if(cur2 !== null) cur2 = cur2.next;
        
    }
   
    
    return res.next;
};

File: solutions/002.add-two-numbers/add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode result = new ListNode(0);
        ListNode head1 = l1, head2 = l2;
        ListNode temp = result;
        int carry = 0;
        while(head1!=null || head2!=null){
            int sum = (head1!=null?head1.val:0)+ (head2!=null?head2.val:0)+carry ;
            temp.next = new ListNode(sum%10);
            carry = sum / 10;
            temp = temp.next;
            if(head1!=null) head1 =head1.next;
            if(head2!=null) head2 = head2.next;
        }
        if(carry != 0){
            temp.next = new ListNode(carry);
        }

        return result.next;
        
        
        
    }
}